### PR TITLE
Update deployment.md

### PR DESCRIPTION
**Pull request by yanshiyason**
Add instructions for docker

### CODE REVIEW

File: guides/deployment.md [hunk 1]
```diff
@@ -14,3 +14,19 @@ For example:
...
+
+Make sure rust is installed prior to running `mix deps.compile`. You can see examples of what commands to include in your Dockerfile by looking at the official rust Dockerfiles. For example, here are the commands for [`alpine3.11`](https://github.com/rust-lang/docker-rust/blob/009cc0a821ff773d54875350312731ed490d5cce/1.43.1/alpine3.11/Dockerfile) based images.
+
+If your Dockerfile is separated into a `build stage` and a `release stage` rust only needs to be installed during the build phase. **However**, your image will need to have `libcc` installed during the `release stage`.
+
+```
+RUN apk add --no-cache libgcc
+```
+
+You will also need to have the folloing environment variable set during the `build` stage or else `mix compile` will fail.
```
Comment (mischov) [inline guides/deployment.md:28]: "following", not "folloing"

File: guides/deployment.md [hunk 2]
```diff
@@ -14,3 +14,19 @@ For example:
 heroku buildpacks:add -i 1 https://github.com/emk/heroku-buildpack-rust.git
 echo "RUST_SKIP_BUILD=1" > RustConfig
 ```
+
+### Deploying with Docker
+
+Make sure rust is installed prior to running `mix deps.compile`. You can see examples of what commands to include in your Dockerfile by looking at the official rust Dockerfiles. For example, here are the commands for [`alpine3.11`](https://github.com/rust-lang/docker-rust/blob/009cc0a821ff773d54875350312731ed490d5cce/1.43.1/alpine3.11/Dockerfile) based images.
+
+If your Dockerfile is separated into a `build stage` and a `release stage` rust only needs to be installed during the build phase. **However**, your image will need to have `libcc` installed during the `release stage`.
```
Comment (mischov) [inline guides/deployment.md:22]: `libgcc`, not `libcc`

File: guides/deployment.md [hunk 3]
```diff
@@ -14,3 +14,19 @@ For example:
...
+
```
Comment (mischov) [inline guides/deployment.md:23]: For example, Alpine does not have `libgcc` installed by default, so it needs to be added with:

Comment (yanshiyason) [reply guides/deployment.md:23]: Sorry for the delay!! I just noticed! Thank you for fixing and improving the PR!!

Comment (mischov) [reply guides/deployment.md:23]: Not a problem, thank you for figuring this out and writing it up in the first place.

**Pull request merged**